lib/rspec/mocks/object_reference.rb

```diff
@@ -42,44 +42,67 @@ def self.anonymous_module?(mod)
     # Used when an object is passed to `object_double`.
     # Represents a reference to that object.
     #
-    # @private
+    # @param object
     class DirectObjectReference
       def initialize(object)
         @object = object
       end
 
+      # The objects desription (`#inspect`).
+      # @return String
       def description
         @object.inspect
       end
 
+      # Not relevant to a direct object reference.
       def const_to_replace
         raise ArgumentError,
               "Can not perform constant replacement with an object."
       end
 
+      # The target of the verifying double (the object's class).
+      #
+      # @return Class
       def target
         @object.class
       end
```

Comment:
> I don't think so, because we expect a class from it I think...

But `NamedObjectReference#target` may not return a class.  Example:

``` ruby
class Application
  LOGGER = Logger.new(...)
end

# in a spec
object_double("Application::Logger").as_stubbed_const
```

In this case `object_double` is using a `NamedObjectReference` and `NamedObjectReference#target` will return the original value of `Application::LOGGER`, which is a `Logger` instance, not a class.  In contrast, the reference created by `object_double(Application::LOGGER)` will be a `DirectObjectReference` and will return the `Logger` class from `#target`.  Seems odd/inconsistent that calling `#target` on the references created by `object_double(Application::LOGGER)` and `object_double("Application::LOGGER")` gets you different things.

Also, `DirectObjectReference` currently provides no method to get the object it is referencing, which seems extremely odd.